### DISCUSSION TITLE
Monogame Question about Texture2D object being interchanged with images of the same dimension in pixels?

### DISCUSSION THREAD

**Discussion by PicoAnimate** [MonoGame/MonoGame#8143] [Content]
I have several images which are the same dimension in pixels, 96 X 96. Each image is loaded into a Texture2d Object as illustrated below on lines 4, 5, and 6.

I have noticed that I can use my texture object to point to either the First, Second or "ManWithOneHandRaised" image as illustrated on lines 7 and 8.

Is it okay, to have a texture2d object point to other objects of the same size? It seems to work perfectly?

// Note, that all image
1.   Texture2D FirstImage; 
2.  Texture2D SecondImage;
3.  Texture2d AGeneralImage; 
4.   AGeneralImage = Content.Load<Texture2D>("ManWithOneHandRaised");
5.   FirstImage =  Content.Load<Texture2D>("ManStandingRightSideUp");
6.   SecondImage = Content.Load<Texture2D>("ManStandingOnHisHands");

7. AGeneralImage = FirstImage;

8. AGeneralImage = SecondImage;

**Comment by ThomasFOG** [comment]
Yes, in C#, variables pointing to an object instance are pointers and do not carry any properties beside their memory address.

You might want to [learn about what pointers are](https://www.tutorialsteacher.com/csharp/csharp-value-type-and-reference-type), and what is the difference between "value types" and "reference types".

(I grabbed the first link I found, likely not the best resource to learn. I would recommend a good C# book such as [C# 12 in a Nutshell: The Definitive Reference](https://www.amazon.com/C-12-Nutshell-Definitive-Reference/dp/1098147448).)

**Reply by PicoAnimate** [reply]
Thanks mrHelmut, I appreciate the advice. I know very well what a pointer, a reference type and value type are. Before working with .NET in 2002, I worked with C, C++, and assembler. 

I have tried pointing different Texture2d objects at different textures that are not the same size in Pixels. Yet the pointer still seemed to work just fine. I just wanted to be sure, about what I was witnessing in the compiler. When I deploy my app on mobile devices I don't want to have any surprises, even if the code passes all of my test cases.

**Reply by ThomasFOG** [reply]
Size and content of allocated objects of the same type are inconsequential to pointers. If something fails, it is not due to memory or the language, but due to how you'd handle sizes down the line.

**Reply by PicoAnimate** [reply]
Thank you !!!
